multicluster/cmd/multicluster-controller/controller.go

```diff
@@ -92,29 +98,26 @@ func getCaConfig(controllerNs string) *certificate.CAConfig {
 	return &certificate.CAConfig{
 		CAConfigMapName: configMapName,
 		// the key pair name has to be "tls" https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L221
-		PairName:           "tls",
-		CertDir:            certDir,
-		ServiceName:        serviceName,
-		SelfSignedCertDir:  selfSignedCertDir,
-		MutationWebhooks:   getMutationWebhooks(controllerNs),
-		ValidatingWebhooks: getValidationWebhooks(controllerNs),
-		CertReadyTimeout:   2 * time.Minute,
-		MaxRotateDuration:  time.Hour * (24 * 365),
+		PairName:                  "tls",
+		CertDir:                   certDir,
+		ServiceName:               serviceName,
+		SelfSignedCertDir:         selfSignedCertDir,
+		MutationWebhookSelector:   getWebhookLabel(controllerNs),
+		ValidatingWebhookSelector: getWebhookLabel(controllerNs),
+		CertReadyTimeout:          2 * time.Minute,
+		MaxRotateDuration:         time.Hour * (24 * 365),
 	}
 }
 
-func getValidationWebhooks(controllerNs string) []string {
-	if controllerNs != "" {
-		return []string{fmt.Sprintf(validationWebhooksNamePattern, controllerNs, "-")}
+func getWebhookLabel(controllerNs string) *metav1.LabelSelector {
+	labels := mcDefaultServedLabels
+	if len(controllerNs) > 0 {
+		// Leader controller has an additional label "served-in: $controllerNs" on the Webhooks.
+		labels["served-in"] = controllerNs
```

Comment:
Is this label needed to uniquely identify the webhook?

Reply:
Yes, having offline sync with @luolanzone , this label is also needed by mc-controller in the leader cluster to identify the webhooks for updating CA bundle.